### PR TITLE
Add tests to enterprise fees w tax report by producer

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

- Related to #6844 / #10304 (built on top of this PR)

This PR aims to cover the specifications in #6844 and cover for the failing test cases, spotted during manual testing.  

**Pending test cases**

1) When an order contains items from two different suppliers (each with their enterprise fees set on incoming exchanges), the report should split the taxes by supplier; it currently displays all fees regardless of the fee applicability, which leads to incorrect totals

:orange_circle: `Pending` - Order with line one line item from each supplier (1, 2):

image

2) Supplier filter returns no results

:orange_circle: `Pending` - Selecting a supplier on the supplier filter returns no results:

image

Other added test cases:

:green_circle: Order with 1 line item from supplier (1):

image

:green_circle: Order with one line item from supplier (2):

image

**Some other comments**

- In terms of access, I believe the report should be accessible as:
  - superadmin
  - distributor/coordinator of the OC -> I've set this to be the default role of the spec
  - suppliers - although I think `supplier A` should not be able to access the data from `supplier B`

I'm not sure we need a system test for this. Maybe good enough to have a request spec covering this.

#### What should we test?

- to be cherry picked in the context of #10304

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by filipefurtad0** [comment]
To illustrate the current issue with the report, it's useful to notice that the commit https://github.com/openfoodfoundation/openfoodnetwork/pull/10797/commits/05e94b529f6be842166320b23be3bcbb2d399262 adds a second incoming exchange, from an additional supplier (`supplier2`):

image

This commit does not change the order, keeping one line item only, from the first supplier:

```
    before do
      order.line_items.create({ variant: variant, quantity: 1, price: 100 })
      order.update!({
                      order_cycle_id: order_cycle.id,
                      ship_address_id: ship_address.id
                    })
      # This will load the enterprise fees from the order cycle.
      # This is needed because the order instance was created
      # independently of the order_cycle.
      order.recreate_all_fees!
      while !order.completed?
        break unless order.next!
      end
    end
```

Comparing, when rendering the report - before this commit:

image

After this commit:
image

File: spec/system/admin/reports/enterprise_summary_fees/enterprise_summary_fee_with_tax_report_by_producer_spec.rb [hunk 1]
```diff
@@ -31,17 +31,20 @@
...
+  let!(:distributor){ create(:distributor_enterprise_with_tax, name: 'Distributor') }
   let!(:payment_method){ create(:payment_method, :flat_rate) }
   let!(:shipping_method){ create(:shipping_method, :flat_rate) }
 
   let!(:order){ create(:order_with_distributor, distributor: distributor) }
   let!(:order_cycle){
-    create(:simple_order_cycle, suppliers: [supplier], distributors: [distributor],
-                                variants: [variant], name: "oc1")
+    create(:simple_order_cycle, suppliers: [supplier, supplier2], distributors: [distributor],
+                                variants: [variant, variant2], name: "oc1")
```
Comment (filipefurtad0) [inline spec/system/admin/reports/enterprise_summary_fees/enterprise_summary_fee_with_tax_report_by_producer_spec.rb]: I finally found out the issue: adding the variants to the order cycle in this manner created exchanges in such a way, that enterprise fees were being applied to both incoming exchanges, regardless of the supplier. 

This was quite intriguing, as I could not spot any difference in the DB... Anyway, it's not a bug, and a set-up is proposed in the next commit (to be squashed with this one).

File: spec/system/admin/reports/enterprise_summary_fees/enterprise_summary_fee_with_tax_report_by_producer_spec.rb [hunk 2]
```diff
@@ -31,54 +31,87 @@
...
+  let!(:order){ create(:order_with_distributor, distributor: distributor) }
   before do
     order_cycle.coordinator_fees << coordinator_fees
     order_cycle.exchanges.incoming.first.exchange_fees.create!(enterprise_fee: supplier_fees)
+    order_cycle.exchanges.incoming.first.exchange_variants.create!(variant: variant)
     order_cycle.exchanges.incoming.second.exchange_fees.create!(enterprise_fee: supplier_fees2)
+    order_cycle.exchanges.incoming.second.exchange_variants.create!(variant: variant2)
     order_cycle.exchanges.outgoing.first.exchange_fees.create!(enterprise_fee: distributor_fee)
+    order_cycle.exchanges.outgoing.first.exchange_variants.create!(variant: variant)
+    order_cycle.exchanges.outgoing.first.exchange_variants.create!(variant: variant2)
```
Comment (filipefurtad0) [inline spec/system/admin/reports/enterprise_summary_fees/enterprise_summary_fee_with_tax_report_by_producer_spec.rb:114]: Ping @dacook, from our chat earlier on: this setup works correctly, as opposed to the changes I've introduced on the [previous commit](https://github.com/openfoodfoundation/openfoodnetwork/pull/10797/commits/bd8e7404192aebc75326165163a3792095eaf743#r1185335232).

**Comment by filipefurtad0** [comment]
Hey @dacook ,

I'm keeping this as a draft, as there still are a couple of things to address (`included tax` section, general refactoring), but I'd say the pending test cases 1) and 2) are good to work on. I hope this helps.

Please let me know if you have any comments or suggestions on my findings and approach to the tests.

**Comment by filipefurtad0** [comment]
Hey @dacook - I've updated the spec to include tests on the `inclusive test` section. I think the Total inclusive tax value is not correct, red line below: 

image

So, this makes up the 3rd section as `pending("test case (3), see #10797")`, in the spec.

**Comment by dacook** [comment]
Thank you!

**Comment by abdellani** [comment]
> 2. Supplier filter returns no results
> 
> orange_circle `Pending` - Selecting a supplier on the supplier filter returns no results:
> 
> image
> 
> Other added test cases:
> 

This issue is fixed on https://github.com/openfoodfoundation/openfoodnetwork/pull/10304

**Pull request closed without merging**